json/src/main/scala/io/sphere/json/FromJSON.scala

```diff
@@ -170,17 +170,43 @@ object FromJSON {
 
   implicit val moneyReader: FromJSON[Money] = new FromJSON[Money] {
     override val fields = Set("centAmount", "currencyCode")
-    def read(jval: JValue): JValidation[Money] = jval match {
-      case o: JObject =>
-        (field[Currency]("currencyCode")(o),
-        field[Long]("centAmount")(o)).mapN(mkMoney)
-      case _ => fail("JSON object expected.")
+
+    def read(value: JValue): JValidation[Money] = value match {
+      case o: JObject ⇒
+        (field[Long]("centAmount")(o), field[Currency]("currencyCode")(o)).mapN(
+          Money.fromCentAmount)
+
+      case _ ⇒ fail("JSON object expected.")
+    }
+  }
+
+  implicit val highPrecisionMoneyReader: FromJSON[HighPrecisionMoney] = new FromJSON[HighPrecisionMoney] {
+    override val fields = Set("preciseAmount", "currencyCode", "fractionDigits")
+
+    import cats.implicits._
+    
+    def read(value: JValue): JValidation[HighPrecisionMoney] = value match {
+      case o: JObject ⇒
+        (field[Long]("preciseAmount")(o),
+          field[Int]("fractionDigits")(o),
+          field[Currency]("currencyCode")(o),
+          field[Option[Long]]("centAmount")(o)).mapN(HighPrecisionMoney.fromPreciseAmount)
```

Comment:
Why do we need to read `centAmount`. I though this is a lazily calculated value for output only not to break everything?

Reply:
I described its purpose in the other comment: https://github.com/sphereio/sphere-scala-libs/pull/34#discussion_r170938905

Does this explanation help?

Reply:
yes and no.
I understand why you want it, but I think it's dangerous.
Somebody can send a json containing `preciseAmount` and a totally different `centAmount`.
And that totally wrong `centAmount` could be used by some tools.